bionic/persistence.py

```diff
@@ -149,27 +149,50 @@ def load_artifact(self):
         Returns the nearest cached artifact for this provenance, if one exists.
         """
 
+        try:
+            entry = self._get_nearest_entry_with_artifact()
+
+            if entry is None:
+                return None
+
+            return entry.artifact
+
+        except InternalCacheStateError as e:
+            self.raise_state_error_with_explanation(e)
+
+    def load_local_artifact(self):
```

Comment:
I think this method is doing more than just `load`. Naming is hard. Wondering if having something like `sync` in the name would make things clearer?

Reply:
Yes, I think that's a good idea.